kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt

```diff
@@ -317,20 +319,24 @@ class TunnelService : VpnService() {
             tunnelState = State.CONNECTING
             updateStatusNotification(TunnelStatusNotification.Connecting)
 
-            connlibSessionPtr =
-                ConnlibSession.connect(
-                    apiUrl = config.apiUrl,
-                    token = token,
-                    deviceId = deviceId(),
-                    deviceName = getDeviceName(),
-                    osVersion = Build.VERSION.RELEASE,
-                    logDir = getLogDir(),
-                    logFilter = config.logFilter,
-                    callback = callback,
-                )
-
-            startNetworkMonitoring()
-            startDisconnectMonitoring()
+            val executor = Executors.newSingleThreadExecutor()
+
+            executor.execute {
+                connlibSessionPtr =
+                    ConnlibSession.connect(
+                        apiUrl = config.apiUrl,
+                        token = token,
+                        deviceId = Tasks.await(FirebaseInstallations.getInstance().id),
```

Comment:
Can we read that earlier? I.e. when Firezone first starts and let it crash if it doesn't work?

Reply:
This can fail in certain very edge cases and I would rather avoid crashing then